File: 05.OOP/EncapsulationAndPolymorphism/Shapes/Rectangle.cs
namespace Shapes
{
    public class Rectangle : BasicShape, IShape
    {
        public Rectangle(double width, double height)
            : base(width, height)
        {

        }

        public override double CalculateArea()
        {
            double area = this.Width * this.Height;
            return area;
        }

        public override double CalculatePerimeter()
        {
            double perimeter = 2 * this.Width + 2 * this.Height;
            return perimeter;
        }
    }
}

File: 05.OOP/OOPInJava/src/Geometry/shapes/Rectangle.java
package Geometry.shapes;

import Geometry.vertices.Vertex2D;

public class Rectangle extends PlaneShape {
	
	private double width;
	private double height;

	public Rectangle(Vertex2D startPoint, double width, double height) {
		super(new Vertex2D[]{startPoint});
		this.setHeight(height);
		this.setWidth(width);
	}

	public double getWidth() {
		return width;
	}

	public void setWidth(double width) {
		this.width = width;
	}

	public double getHeight() {
		return height;
	}

	public void setHeight(double height) {
		this.height = height;
	}

	@Override
	public double getPerimeter() {
		double perimeter = 2 * this.getWidth() * this.getHeight();
		return perimeter;
	}

	@Override
	public double getArea() {
		double area = this.getWidth() * this.getHeight();
		return area;
	}
}
